pkg/validation/spec/gnostic.go

```diff
@@ -0,0 +1,1554 @@
+	}
+	ok = true
+
+	switch p := g.Oneof.(type) {
+	case *openapi_v2.SchemaItem_FileSchema:
+		fileSchema := p.FileSchema
+
+		if nok, err := k.VendorExtensible.FromGnostic(fileSchema.VendorExtension); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+
+		k.Format = fileSchema.Format
+		k.Title = fileSchema.Title
+		k.Description = fileSchema.Description
+		k.Required = fileSchema.Required
+		k.Type = []string{fileSchema.Type}
+		k.ReadOnly = fileSchema.ReadOnly
+
+		if fileSchema.ExternalDocs != nil {
+			k.ExternalDocs = &ExternalDocumentation{}
+			if nok, err := k.ExternalDocs.FromGnostic(fileSchema.ExternalDocs); err != nil {
+				return false, err
+			} else if !nok {
+				ok = false
+			}
+		}
+
+		if fileSchema.Example != nil {
+			if err := fileSchema.Example.ToRawInfo().Decode(&k.Example); err != nil {
+				return false, err
+			}
+		}
+
+		if fileSchema.Default != nil {
+			if err := fileSchema.Default.ToRawInfo().Decode(&k.Default); err != nil {
+				return false, err
+			}
+		}
+
+	case *openapi_v2.SchemaItem_Schema:
+		schema := p.Schema
+
+		if nok, err := k.FromGnostic(schema); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+	}
+
+	return ok, nil
+}
+
+// SecurityDefinitions
+
+func (k SecurityDefinitions) FromGnostic(g *openapi_v2.SecurityDefinitions) (ok bool, err error) {
+	ok = true
+	for _, v := range g.GetAdditionalProperties() {
+		if v == nil {
+			continue
+		}
+		secScheme := &SecurityScheme{}
+		if nok, err := secScheme.FromGnostic(v.Value); err != nil {
+			return false, err
+		} else if !nok {
+			ok = false
+		}
+		k[v.Name] = secScheme
+	}
+
+	return ok, nil
+}
+
+type GnosticCommonSecurityDefinition interface {
+	GetType() string
+	GetDescription() string
+}
+
+func (k *SecuritySchemeProps) FromGnostic(g GnosticCommonSecurityDefinition) (ok bool, err error) {
+	ok = true
+	k.Type = g.GetType()
+	k.Description = g.GetDescription()
+
+	if hasName, success := g.(interface{ GetName() string }); success {
+		k.Name = hasName.GetName()
+	}
+
+	if hasIn, success := g.(interface{ GetIn() string }); success {
+		k.In = hasIn.GetIn()
+	}
+
+	if hasFlow, success := g.(interface{ GetFlow() string }); success {
+		k.Flow = hasFlow.GetFlow()
+	}
+
+	if hasAuthURL, success := g.(interface{ GetAuthorizationUrl() string }); success {
+		k.AuthorizationURL = hasAuthURL.GetAuthorizationUrl()
+	}
+
+	if hasTokenURL, success := g.(interface{ GetTokenUrl() string }); success {
+		k.TokenURL = hasTokenURL.GetTokenUrl()
+	}
+
+	if hasScopes, success := g.(interface {
+		GetScopes() *openapi_v2.Oauth2Scopes
+	}); success {
+		scopes := hasScopes.GetScopes()
+		if scopes != nil {
+			k.Scopes = make(map[string]string, len(scopes.AdditionalProperties))
+			for _, v := range scopes.AdditionalProperties {
+				k.Scopes[v.Name] = v.Value
+			}
+		}
+	}
+
+	return ok, nil
+}
+
+func (k *SecurityScheme) FromGnostic(g *openapi_v2.SecurityDefinitionsItem) (ok bool, err error) {
```

Comment:
I think `ok` can be dropped from here once SecuritySchemeProps/VendorExtensible stop returning `ok`